From 922d40b0e8f88ade8a086a85b16c8bd32b4e6ac4 Mon Sep 17 00:00:00 2001
From: Kevin Hao <kexin.hao@windriver.com>
Date: Fri, 14 Nov 2014 10:41:12 +0800
Subject: [PATCH 489/666] memblock: change the annotation to __init_memblock
 for memblock_is_reserved()

This function may be used by modules, so change the annotation to
__init_memblock and also export the symbol.

Signed-off-by: Kevin Hao <kexin.hao@windriver.com>
[Xulin: minor adjustment]
Signed-off-by: Xulin Sun <xulin.sun@windriver.com>
---
 mm/memblock.c | 5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git a/mm/memblock.c b/mm/memblock.c
index 43d0919..0b89f66 100644
--- a/mm/memblock.c
+++ b/mm/memblock.c
@@ -1601,10 +1601,13 @@ static int __init_memblock memblock_search(struct memblock_type *type, phys_addr
 	return -1;
 }
 
-bool __init memblock_is_reserved(phys_addr_t addr)
+bool __init_memblock memblock_is_reserved(phys_addr_t addr)
 {
 	return memblock_search(&memblock.reserved, addr) != -1;
 }
+#ifndef CONFIG_ARCH_DISCARD_MEMBLOCK
+EXPORT_SYMBOL_GPL(memblock_is_reserved);
+#endif
 
 bool __init_memblock memblock_is_memory(phys_addr_t addr)
 {
-- 
2.7.4

